src/Arriba/Arriba/Diagnostics/Tracing/ArribaEventSource.cs

```diff
@@ -63,9 +64,20 @@ public interface IArribaEvents : IEventSource
 
     public sealed class ArribaEventSource : EventSource, IArribaEvents
     {
+
+        public delegate  void TimedFunction();
+        
         public static IArribaEvents Log { get; }
         public static IArribaServiceEvents Service { get; }
 
+
+        public static Task<TimeSpan> Time(TimedFunction timedFunction)
+        {
+            DateTimeOffset startTime = DateTimeOffset.Now;
+            timedFunction.Invoke();
+            return Task.FromResult<TimeSpan>(DateTimeOffset.Now - startTime);
+        }
```

Comment:
Use a Stopwatch, not date time

https://stackoverflow.com/questions/2143140/c-sharp-datetime-now-precision

Reply:
Updated